python_modules/dagster/dagster/core/execution/execution_results.py

```diff
@@ -103,9 +104,35 @@ def output_values(self) -> Dict[str, Any]:
                             ] = value
                     else:
                         results[step_output_handle.output_name] = value
+        else:
+            raise DagsterInvariantViolationError(
+                "Trying to fetch output values, but output_capturing_enabled was False for the run"
+            )
 
         return results
 
+    def output_value(self, output_name=DEFAULT_OUTPUT):
+        """Get a computed output value.
+
+        Note that calling this method will reconstruct the pipeline context (including, e.g.,
+        resources) to retrieve materialized output values.
+
+        Args:
+            output_name(str): The output name for which to retrieve the value. (default: 'result')
+
+        Returns:
+            Union[None, Any, Dict[str, Any]]: ``None`` if execution did not succeed, the output value
+                in the normal case, and a dict of mapping keys to values in the mapped case.
+        """
+        check.str_param(output_name, "output_name")
+
+        if not self._output_capture:
+            raise DagsterInvariantViolationError(
+                "Trying to fetch output values, but output_capturing_enabled was False for the run"
+            )
+
+        return self._output_capture[StepOutputHandle(str(self.handle), output_name)]
```

Comment:
I don't think this will work for the dynamic case

Reply:
Good call - just updated.